woocommerce/woocommerce-gateway-stripe
# Remove key sharing fields and update to the Account keys modal UI

Part of https://github.com/woocommerce/woocommerce-gateway-stripe/issues/3196

## Changes proposed in this Pull Request:

This PR lays a lot of the groundwork for removing the publishable and secret key fields in the Stripe settings and instead replacing it with a link/button to connect via the Stripe oauth flow. 

https://github.com/user-attachments/assets/b9beaf80-ebcb-4de0-b34e-eea586892196

This PR is only part of the work involved in #3196. This PR:

1. Removes the publishable and secret key fields. 
2. Adds interim designs for the connect flow. 
3. As part of this PR I've made a number of other changes. Like:
    - Introducing the Chip component. Based on the designs in Figma (Kofpo8jj4zDWXDsVdlSz0f-fi-1748_72562) ([screenshot](https://github.com/user-attachments/assets/d5aaae45-afc3-4720-83f9-c3725bde4bfc))
    - Updating our `SectionStatus` component which was largely based on this Chip component.
    - Adding all the WP color variables we were missing like `$wp-yellow-30`.
    - Adding assets for the Stripe mark and Woo logo.
    - I've refactored this keys modal so the UI elements are more separated into their own files. This should help with readability and maintenance. 
    - There's a couple of other things but I'll mention them inline. 

## Testing instructions

1. Checkout this branch. 
2. Run `npm run build` to compile the JS assets. 
3. Go to **WooCommerce > Settings > Payments > Stripe > Settings (tab)**
4. Click on the **Edit account keys** button. 
5. You should see the modal above. 

**VERIFY STATE UPDATES**

**WEBHOOKS**

1. Change between Live and Test tabs and confirm that the content updates accordingly. 
2. If you have previously configured webhooks via the button added in 8.5.0 (https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3209) you should see:
    - The text at the bottom of the modal should include your site's webhook URL (`Your webhooks are configured and will be sent to: https://wcdev.local/?wc-api=wc_stripe.`)
    - The Webhook status should say "Configured".
3. If you had previously just entered your webhook secret manually you should see:
    - The text at the bottom will say: `Webhooks have been manually configured via a webhook secret.`.
    - The Webhook status should say "Enabled".
4. If you delete all webhook keys/data (see code snippet below), you should see: 
    - The text at the bottom will say: `Configuring webhooks will enable your store to receive notifications on charge statuses from Stripe.`
    - The Webhook status will be red and say "Disabled"

```php
$settings = get_option( 'woocommerce_stripe_settings' );
unset( $settings['webhook_data'] );
unset( $settings['test_webhook_data'] );
unset( $settings['test_webhook_secret'] );
update_option( 'woocommerce_stripe_settings', $settings );
```

**ACCOUNT**

1. If you have keys set, the Account label will say "Incomplete". This will change once we've implemented the oauth flow that allows merchants to connect their account to the WooCommerce App.
    - There's also a tooltip that explains what "incomplete" means. 

<p align="center">
<img width="305" alt="Screenshot 2024-07-12 at 12 38 55 PM" src="https://github.com/user-attachments/assets/d7cce71a-c102-45db-96d8-ac003d11944b">
</p> 

2. If you have no keys set at all. The Account label will be read and say "Disconnected". The configure webhooks buttons should also not be shown.

<p align="center">
<img width="500" alt="Screenshot 2024-07-19 at 10 53 28 AM" src="https://github.com/user-attachments/assets/a588dc5f-3296-4e8b-8532-37c3ee481f5d">
</p>

---

**KNOWN ISSUES**
There are a number of known issues with this PR as it is now. We'll follow up in other PRs to fix these. These have all be listed on the Epic - https://github.com/woocommerce/woocommerce-gateway-stripe/issues/3196. 

- The oauth URL used in the connect button is the live link no matter if you're on the live or test tab. 
- Webhooks are supposed to be automatically configured after connecting. We'll follow up with that separately. 
- The [description text](https://github.com/user-attachments/assets/18b5961e-0cf9-4b75-88bc-6589421f4409) is just a copy of the other connect flow text. We may want to change this to be more related to this context.

---


**Post merge**
